app/actions/websocket/integrations.ts

```diff
@@ -0,0 +1,16 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+import IntegrationsManager from '@init/integrations_manager';
+
+export function handleOpenDialogEvent(serverUrl: string, msg: WebSocketMessage) {
+    const data: string = msg.data?.dialog;
+    if (!data) {
+        return;
+    }
+    try {
+        const dialog: InteractiveDialogConfig = JSON.parse(data);
+        IntegrationsManager.getManager(serverUrl).setDialog(dialog);
```

Comment:
We could check the server, so dialogs from other servers are not show, but should not be needed because:

- The server won't send a dialog for a expired trigger id
- Trigger ids expire fast (3-5 seconds if I remember correctly)
- Trigger ids are created only because of a user action.

The chance of receiving a dialog from a different server is really low (but never 0). Thoughts?

Reply:
another check to ensure that it does not show up is not a bad thing